### DISCUSSION TITLE
How to load an Pillow opened image into `image-rs`

### DISCUSSION THREAD

**Discussion by baseplate-admin** [PyO3/pyo3#3090] [Questions] [answered]
Hi thanks for creating `pyo3`,

So i have an image opened with `PIllow` and i would like to transfer the image from Python to Rust and then process it in `Image-rs` library.

Is it possible to do so without saving the file to the disk and then loading it with `rust` ( I was thinking of loading the memory directly and somehow process it? )?

**Comment by davidhewitt** [comment, marked as answer]
Just use a `BytesIO` or similar with `Image.save` to keep the image in-memory and then pass the bytes to Rust as a `#[pyfunction]` input to load however you like on the Rust side. This isn't any different to how you might share an in-memory image with other Python libraries.

https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.save
